Rustdoc allows multiple declarations of a feature

I tried this code:

```rust
#![feature(doc_cfg)]
#![feature(doc_cfg)]

fn main() {
    println!("Hello, world!");
}
```

I expected to see this happen: Both `cargo build` and `cargo doc` to fail because of multiple declarations of the `doc_cfg` feature.

Instead, this happened: `cargo build` failed while `cargo doc` succeeded:

```console
> cargo build
   Compiling foo v0.1.0 (/tmp/tmp.NidTpZ70xE/foo)
error[E0636]: the feature `doc_cfg` has already been declared
 --> src/main.rs:2:12
  |
2 | #![feature(doc_cfg)]
  |            ^^^^^^^

error: aborting due to previous error

For more information about this error, try `rustc --explain E0636`.
error: could not compile `foo`

To learn more, run the command again with --verbose.

> cargo doc
 Documenting foo v0.1.0 (/tmp/tmp.NidTpZ70xE/foo)
    Finished dev [unoptimized + debuginfo] target(s) in 0.48s
```

### Meta
`rustc --version --verbose`:
```
rustc 1.52.0-nightly (07194ffcd 2021-02-10)
binary: rustc
commit-hash: 07194ffcd25b0871ce560b9f702e52db27ac9f77
commit-date: 2021-02-10
host: x86_64-unknown-linux-gnu
release: 1.52.0-nightly
LLVM version: 11.0.1
```

--- a/library/std/src/os/mod.rs
+++ b/library/std/src/os/mod.rs
@@ -23,6 +23,7 @@ pub use crate::sys::windows_ext as windows;
 pub mod linux;
 
 #[cfg(doc)]
+#[stable(feature = "wasi_ext_doc", since = "1.35.0")]
 pub use crate::sys::wasi_ext as wasi;
 
 // If we're not documenting libstd then we just expose the main modules as we otherwise would.

--- a/library/std/src/sys/mod.rs
+++ b/library/std/src/sys/mod.rs
@@ -111,7 +111,7 @@ cfg_if::cfg_if! {
 cfg_if::cfg_if! {
     if #[cfg(target_os = "wasi")] {
         // On WASI we'll document what's already available
-        #[stable(feature = "rust1", since = "1.0.0")]
+        #[stable(feature = "wasi_ext_doc", since = "1.35.0")]
         pub use self::ext as wasi_ext;
     } else if #[cfg(any(target_os = "hermit",
                         target_arch = "wasm32",
@@ -125,6 +125,7 @@ cfg_if::cfg_if! {
     } else {
         // On other platforms like Windows document the bare bones of WASI
         #[path = "wasi/ext/mod.rs"]
+        #[stable(feature = "wasi_ext_doc", since = "1.35.0")]
         pub mod wasi_ext;
     }
 }

--- a/src/librustdoc/core.rs
+++ b/src/librustdoc/core.rs
@@ -492,6 +492,7 @@ crate fn run_global_ctxt(
             tcx.ensure().check_mod_attrs(module);
         }
     });
+    rustc_passes::stability::check_unused_or_stable_features(tcx);
 
     let access_levels = tcx.privacy_access_levels(LOCAL_CRATE);
     // Convert from a HirId set to a DefId set since we don't always have easy access

--- a/src/librustdoc/lib.rs
+++ b/src/librustdoc/lib.rs
@@ -49,6 +49,7 @@ extern crate rustc_metadata;
 extern crate rustc_middle;
 extern crate rustc_mir;
 extern crate rustc_parse;
+extern crate rustc_passes;
 extern crate rustc_resolve;
 extern crate rustc_session;
 extern crate rustc_span as rustc_span;

--- a/src/test/rustdoc-ui/rustc-check-passes.rs
+++ b/src/test/rustdoc-ui/rustc-check-passes.rs
@@ -1,0 +1,4 @@
+#![feature(box_syntax)]
+#![feature(box_syntax)] //~ ERROR
+
+pub fn foo() {}

--- a/src/test/rustdoc-ui/rustc-check-passes.stderr
+++ b/src/test/rustdoc-ui/rustc-check-passes.stderr
@@ -1,0 +1,9 @@
+error[E0636]: the feature `box_syntax` has already been declared
+  --> $DIR/rustc-check-passes.rs:2:12
+   |
+LL | #![feature(box_syntax)]
+   |            ^^^^^^^^^^
+
+error: aborting due to previous error
+
+For more information about this error, try `rustc --explain E0636`.

--- a/src/test/rustdoc/implementor-stable-version.rs
+++ b/src/test/rustdoc/implementor-stable-version.rs
@@ -1,3 +1,4 @@
+#![stable(feature = "bar", since = "OLD 1.0")]
 #![crate_name = "foo"]
 
 #![feature(staged_api)]
@@ -8,6 +9,7 @@ pub trait Bar {}
 #[stable(feature = "baz", since = "OLD 1.0")]
 pub trait Baz {}
 
+#[stable(feature = "baz", since = "OLD 1.0")]
 pub struct Foo;
 
 // @has foo/trait.Bar.html '//div[@id="implementors-list"]//span[@class="since"]' 'NEW 2.0'

